docs/proposals/0683-epp-architecture-proposal/interfaces/interface.go

```diff
@@ -0,0 +1,123 @@
+}
+
+type Scheduler interface {
+	Plugin
+	// ProfileSelection selects scheduling profiles through the implemented
+	// logic, and returns a subset of the registered scheduling profiles.
+	ProfileSelection() map[string]SchedulingProfile
+
+	// SchedulingProfiles lists all of the scheduling profiles registered
+	// with the scheduler.
+	SchedulingProfiles() map[string]SchedulingProfile
+
+	// SchedulingResult takes the output of the result(s) of the scheduling cycle(s)
+	// and makes sense of the data to be consumed by request control.
+	// For example: suppose you have 2 profiles ShadowBoxing Profile & Production Profile.
+	// SchedulingResult would know to simply log the result of ShadowBoxing
+	// profile, and do nothing else with it.
+	SchedulingResult(map[string][]Endpoint) SchedulingResult
+}
+
+// SchedulingProfile is an interface to used to describe a profile that will
+// run for a given scheduling cycle.
+type SchedulingProfile interface {
+	Plugin
+	// PreSchedulePlugins are optional, and will be ran at the start of a
+	// scheduling cycle. This should be scoped to any foundational work needed
+	// that is custom to this scheduling profile.
+	PreSchedulePlugins() []PreSchedule
+	// Filters lists all Filter plugins associated with this Profile. Filters
+	// are optional.
+	Filters() []Filter
+	// Scorers lists all Score plugins associated with this Profile. At
+	// least 1 scorer must be registered for a profile to be valid.
+	Scorers() map[Scorer]int
+	// Selection returns the function that picks the endpoint(s).
+	Selection() Picker
+	// PostSchedulePlugins lists all Filter plugins associated with this
+	// Profile. PostSchedulePlugins are ran after every scheduling cycle,
+	// and are optional.
+	PostSchedulePlugins() []PostSchedule
+}
+
+// Preschedule will be ran at the start of a scheduling cycle. This should be
+// scoped to any foundational work needed that is custom to this scheduling
+// profile.
+type PreSchedule interface {
+	Plugin
+	PreSchedule(ctx context.Context, state scheduling.CycleState, endpoints []Endpoint)
+}
+
+// Filter runs before any scoring, and remove endpoints that are not fit for
+// selection. The framework will return an error to the client if the endpoints
+// are filtered to zero.
+type Filter interface {
+	Plugin
+	Filter(ctx context.Context, state scheduling.CycleState, endpoints []Endpoint) []Endpoint
+}
+
+// Scorer applies a score to each remaining endpoint provided. Scorers SHOULD
+// keep their score values in a normalized range: [0-1]. Any weighting should
+// be added at the SchedulingProfile configuration level.
+type Scorer interface {
+	Plugin
+	Score(ctx context.Context, state scheduling.CycleState, endpoints []Endpoint) []Endpoint
+}
+
+// Picker selects the endpoint(s) from the provided list of scored endpoints.
+// Picker MUST return, one endpoint at minimum.
+type Picker interface {
+	Plugin
+	Selection(ctx context.Context, state scheduling.CycleState, endpoints []Endpoint) []Endpoint
```

Comment:
function names are typically verbs or verb phrases, because they do something. 
`Pick`/`Select` conveys an action: choosing the best endpoint.
so I suggest - 
if we call the interface `Picker` - function should be called `Pick`. 
if we call it `Selector` the function name should be called `Select`